internal/auth0/connection/schema.go

```diff
@@ -853,3 +855,92 @@ var optionsSchema = &schema.Schema{
 		},
 	},
 }
+
+var scimResourceSchema = map[string]*schema.Schema{
+	"connection_id": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "ID of the connection for this SCIM configuration.",
+	},
+	"connection_name": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Name of the connection for this SCIM configuration.",
+	},
+	"strategy": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Schema of the connection for this SCIM configuration.",
+	},
+	"tenant_name": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Name of the tenant for this SCIM configuration.",
+	},
+	"user_id_attribute": {
+		Type:         schema.TypeString,
+		Optional:     true,
+		RequiredWith: []string{"user_id_attribute", "mapping"},
+		Computed:     true,
+		Description:  "User ID attribute for generation unique of user ids. If `user_id_attribute` is set, `mapping` must be set as well. Defaults to `externalId`.",
+	},
+	"mapping": {
```

Comment:
On update both user id and mapping are required, that indicates to use that we should also set these as required.

Reply:
actually, it is somewhat incorrect, it isn't always externalId. I think I will steal Kunal's wording for it...

Reply:
I updated the description to more accurately reflect how it works.